Automattic/wp-calypso
# DocumentHead: Remove setUnreadCount and unreadCount props

See #24977 for background:

> 1. Remove the dedicated `unreadCount` handling from the `DocumentHead` component and state. Arguably, `DocumentHead` should reflect the contents of the HTML head. `unreadCount` is simply reflected in the `<title />`, and it's not a very generic or widely used thing -- I think only the Reader uses it. Hence, I think it's fair to put the burden of managing `unreadCount` state on the consuming component rather than `DocumentHead`.

What's more is that by definition, the `unreadCount` isn't needed on the server side: We only show the `unreadCount` indicator in the title to indicate how many new posts have been published since the reader was first loaded. This is reflected by `<DocumentHead />`'s `unreadCount` prop only having been used in one place -- the `UpdateNotice` component (the orange visual indicator telling the user that there are new posts).

Hence, this PR:
* Changes `UpdateNotice` to handle unread count state internally
* Removes `unreadCount` related `DocumentHead` props and Redux state stuff

To test:
* Verify that Reader still properly shows the correct unread count in the site title (in parens). (Question for Reader folks: What Redux action do we need to dispatch in the console in order to easily simulate new posts?)
* Verify that navigating away properly updates the site title (particularly, that it removes the unread count)
